### ISSUE TITLE
Cross entropy loss produces error 

### ISSUE THREAD

**Issue by ChrKoenig** [citoverse/cito#97]
The "cross-entropy" loss produces an error when trying to fit a multi-class dnn.

**Reprex:**
```
library(cito)

nn.fit <- dnn(Species ~ Sepal.Width + Sepal.Length + Petal.Length + Petal.Width, 
              data = datasets::iris,
              loss = "cross-entropy")
```
```
Error in torch_tensor_cpp(data, dtype, device, requires_grad, pin_memory) : 
  R type not handled
```

**Comment by cmteuffel** [comment]
It's an error thrown by the torch package in tensor.cpp.  Could it be that the case loss = "cross-entropy" is not correctly handled in cito's format_targets function (utils.R)? It seems, the data is not adequately converted into a tensor there. Hence the error.
